students-router.js

```diff
@@ -0,0 +1,88 @@
+const knex = require('knex');
+const knexConfig = require('./knexfile.js');
+const db = knex(knexConfig.development);
+
+const router = require('express').Router();
+
+router.post('/', async (req,res) => {
+    if (!req.body.name || !req.body.cohort_id) {
+        res.status(400).json({error:'missing name / cohort id'});
+    } else {
+        try {
+            const student = await db('students').insert(req.body);
+            res.status(200).json(student);
+        }
+        catch(err) {
+            res.status(500).json({error:err});
+        }
+    }
+
+});
+
+router.get('/', async (req,res) => {
+    try {
+        const students = await db('students');
+        res.status(200).json(students);
+    }
+    catch(err) {
+        res.status(500).json({error:err});
+    }
+});
+
+router.get('/:id', async (req,res) => {
+    const {id} = req.params;
+    try {
+        const student = await
+        db.select('students.id','students.name','cohorts.name as cohort').from('students')
+        .leftJoin('cohorts',{'students.cohort_id':'cohorts.id'})
+        .where({'students.id':id}).first();
+        if (student) {
+            res.status(200).json(student);
+        } else {
+            res.status(404).json({error:'ID not found'});
+        }
+    }
+    catch(err) {
+        res.status(500).json(err);
+    }
+});
+
+router.put('/:id', async (req,res) => {
+    const {id} = req.params;
+    if (!req.body.name || !req.body.cohort_id) {
```

Comment:
this is something I might have missed before when reviewing your code, but for PUT requests, what if the user only should be expected to send _either_ a new name or a new cohort_id? The database will still automatically merge the object you send and what already exists with only one of the necessary properties included, so you could change you OR operator on this line to an AND and it would open the functionality up for that